### DISCUSSION TITLE
Debian for other product of Buffalo

### DISCUSSION THREAD

**Discussion by haconbk** [1000001101000/Debian_on_Intel_Terastations#8] [Ideas]
Hello!

I have a Buffalo WS-Q4.0TL-R5, this product have chipset Intel Atom 1.66 GHz. I lost USB recovery key (to install Windows storage server 2003) and I want to install Debin into my Nas. But this nas haven't VGA ports (see https://ithardwarehub.com/part-no-ws-qv4-0tl-r5-buffalo-terastation-pro-quad-wss-ws-qvl-r5-network-storage-server-intel-atom-1-66-ghz-4-tb-4-x-1-tb-rj-45-network-usb-usb/). I try used ssh but no reply from NAS.

Can't you help me.
Thanks and best regards

**Comment by 1000001101000** [comment]
Good to hear from you!

I didn’t know they shipped these without VGA ports. I also thought the Windows variant needed them for setup/etc….weird.

I do not know any reason that should change how the installer works.

Did you set the switch on the back to USB?
Could you post what the last message on the LCD was?

**Reply by haconbk** [reply]
this device have not switch to USB button

back
![IMG_4586](https://user-images.githubusercontent.com/5661389/155848019-6e944cd3-ed18-488c-a7b9-d8af8b7823c5.JPG)
front (with recovery button)
![IMG_4587](https://user-images.githubusercontent.com/5661389/155848166-84824219-3222-472f-830a-49ca25705135.JPG)

when I try to recovery, I need usb recovery (but lost), and follow this instruction
![system-recovery](https://user-images.githubusercontent.com/5661389/155848161-594236ca-7159-4e87-b61e-35f8cebd7bcf.png)

And the last message on the LCD
![IMG_4588](https://user-images.githubusercontent.com/5661389/155848000-d37660ae-f9bb-467d-812b-24a9a4765e98.JPG)

thanks for reply this my comment

**Reply by 1000001101000** [reply]
Interesting.

On the variation I am familiar with the USB switch just puts USB devices to the front of the boot order, if you try booting the device with the Debian image in a usb port and no drives installed I would expect it to boot into the installer (it would be the only entry in the boot order).

Which image did you download and what size was the resulting file? How did you write it to the USB device?

**Reply by haconbk** [reply]
> Interesting.
> 
> On the variation I am familiar with the USB switch just puts USB devices to the front of the boot order, if you try booting the device with the Debian image in a usb port and no drives installed I would expect it to boot into the installer (it would be the only entry in the boot order).
> 
> Which image did you download and what size was the resulting file? How did you write it to the USB device?

i try, but not work with device.
image i used bullseye, buster and stretch with rufus, WinSetupFromUSB in windows, dd in linux. And only "Booting System .."

**Reply by 1000001101000** [reply]
I have a few thoughts:

1. Does the device have internet access when booting? Debian’s netconsole installer which this is based on starts accessing the Debian repositories before SSH starts and won’t start at all if it fails to reach them.2. 

2. even without ssh I would expect the LCD to change color and display an installer message if it was making it to my installer code (which is earlier than any network issues). 

I’m curious what else the LCD says before that message, If I remember correctly it would display an error and turn red if it didn’t find anything to boot from….but also would usually say some thing like (booting from disk 0) if it did find something.


3. Many people have downloaded the HTML of the github page instead if the installer binaries by mistake. Can you post the size and/or MD5 of the image files you tried?

4. Perhaps USB boot works differently for this variation. You could try is writing the image to a sata drive and trying to boot from that ( in case the issue is with usb boot in general ).

I’ve worked with the boot process on 3 different generations of these intel devices ( ( including a ws-qvhl) and they all worked basically the same. I’d be surprised if there is some “secure boot” or other restriction preventing this, but it is a possibility.

My WS-QVHL has a VGA port on the front, 2x ethernet and the usb/hdd boot switch. Until now every 4+ bay Terastation after the Terastation II that I’ve seen had all those features. I would still assume it’s very much like mine, but wonder what else could be different

**Reply by haconbk** [reply]
I think this device have internet access, I can see the lights is on.
Size and md5 code of the image files:
- build 2022-02-25\ts-bullseye-installer.iso 49.3 MB (51,642,368 bytes) 8868CE3624EAE02D873AF961C8E8C2C0
- build 2022-02-25\ts-buster-installer.iso 48.8 MB (51,146,752 bytes) 7956E3286EB4E926F94FECF33398B64C
- build 2022-02-25\ts-stretch-installer.iso 39.9 MB (41,854,976 bytes) 7D3A3B9CDB5C30D0F4FCB21EF174BB16


I try, writing the image to a sata drive and have error:

![IMG_4589](https://user-images.githubusercontent.com/5661389/155869241-542f3321-7ff9-4d98-ae38-955017eb74b6.JPG)

"POST ERR:02FE"

**Comment by Zucky16** [comment]
Just in case anyone else stumbles upon this, when I recently moved my TS-8VHL to the fantastic Debian release I noticed there is no timeout on the USB ISO boot screen to start the Install process.  The default selection was **Install**, however it would sit there indefinitely until I hit enter.  Lucky for me my unit has VGA video, but someone who has no video, they would need to connect a keyboard and wait say 1 minute or so after boot then hit enter to start the install process.  Wait again maybe 2-3 minutes and they should be able to SSH into their unit and continue the install as noted.

**Reply by 1000001101000** [reply]
Good to hear from you!

It should not do that, I'll take a look when I get a moment.

**Reply by 1000001101000** [reply]
I just tried this with the Bullseye installer on my TS-8VHL and it started the ssh-console without intervention as expected.

Which ISO did you use?

**Reply by Zucky16** [reply]
Appreciate the reply, I have Bullseye installed, funny, since I did use that ISO.
Just boots from USB to an indefinite wait for a selection.
![messages_0 (37)](https://user-images.githubusercontent.com/20763743/189359879-60d9ebac-52f3-4aa0-92eb-3e096e49ef66.jpeg)
I downloaded the ISO on Sep-02-2022.

**Comment by 1000001101000** [comment]
What does the lcd screen on the device say when it’s at that menu

**Reply by Zucky16** [reply]
I'll have to get back to you on that one, taking this sucker offline is a pain, I'm running iSCSI to my VM lab from it and SMB share for Veeam backup from my workstations.
Can't remember if it had the _Installing Debian_ custom message or not at that stage.  I know it did after I hit enter to accept it.

Ok so, I did have time since it was easy for me to shut down my lab stuff.

Here's the boot LCD while the Install message is on the screen indefinitely.
![20220909_132035](https://user-images.githubusercontent.com/20763743/189411242-c32c245e-c1e4-412e-b60e-557c97c7b3d3.jpg)
Then after starting the Install by hitting enter
![20220909_132439](https://user-images.githubusercontent.com/20763743/189411303-2cf6d865-97c6-4ff8-ab4f-9f94776dd098.jpg)
